src/zarr/core/array.py

```diff
@@ -977,9 +977,10 @@ def _iter_chunk_regions(
     @property
     def nbytes(self) -> int:
         """
-        The number of bytes that can be stored in this array.
+        The number of bytes that can be stored in the chunks of this array.
         """
-        return self.nchunks * self.dtype.itemsize
+        # TODO: how can this be meaningful for variable-length types?
+        return int(np.prod(self.shape) * self.dtype.itemsize)
```

Comment:
I'm wondering if it would be worth using the exact same logic we use in `info`:

https://github.com/zarr-developers/zarr-python/blob/5bf7bcfcfe6426bea37df5d0691211f6a71daa39/src/zarr/core/array.py#L1432

Reply:
we should definitely only calculate this value in one place. I will look into changing `nbytes` to use the simpler formula above, and then `info` will use `nbytes`